holidayextras/brands
# UXUI-618 - Adding btn-loading and loading as a class of properties

#### What does this PR do? (please provide any background)
This PR plugs the game left by Bootstrap when they deprecated the button loading features of `buttons.less`. This enables us to have a spinner 'on form submit' for example without compromising the width of the button. 

This change has markup that's dependent on it. So in order to have a 'loading' state on your button you need to have a span wrapping the button text together with an `<i>` tag holding the icon.

#### What tests does this PR have?
None

#### How can this be tested?
`npm link` into Pattern Library.

Or grab the `uxui-617` branch on the `static-site-generator-sites` and follow the steps to get a the `hx-customer-contact` template group running locally. After you've done that, navigate to `details-form.html` and click 'search'. You should see the spinner icon correctly aligned and the text should disappear.

#### Screenshots / Screencast
![screen shot 2016-05-27 at 10 56 19](https://cloud.githubusercontent.com/assets/6357978/15604702/b9c3c47c-23f9-11e6-82ec-d85be5ab34c3.png)
![screen shot 2016-05-27 at 10 56 34](https://cloud.githubusercontent.com/assets/6357978/15604703/b9c7c8b0-23f9-11e6-89fa-60c7bf649da9.png)


#### What gif best describes how you feel about this work?
![](http://media0.giphy.com/media/KLO1hNEky1fMc/giphy.gif)

---


#### Reviewers

**Review 1**

**Review 2**

**Review 3**

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.
